implementations/rust/ockam/ockam/src/error.rs

```diff
@@ -1,43 +1,68 @@
 #![deny(missing_docs)]
 #![allow(missing_docs)] // Contents are self describing for now.
-#[derive(Clone, Copy, Debug)]
+
+use ockam_core::{
+    errcode::{ErrorCode, Kind, Origin},
+    thiserror, Error2,
+};
+
+#[derive(Clone, Copy, Debug, thiserror::Error)]
 pub enum OckamError {
```

Comment:
I suspect the thiserror implementations like this would be better as
```rs
impl core::fmt::Display for OckamError {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        core::fmt::Debug::fmt(self, f)
    }
}

impl ockam_core::compat::Error for OckamError {}
```

I believe this will fix the `no_std` issue. Note that std::error::Error is not available to no_std code currently, so we have a shim for it in the compat module. (I know I told you that thiserror is fine, but I think this probably means we can't use it).

On the bright side, this should be a lot less code.